internal/app/machined/pkg/controllers/k8s/kubelet_spec.go

```diff
@@ -242,7 +242,12 @@ func NewKubeletConfiguration(cfgSpec *k8s.KubeletConfigSpec) (*kubeletconfig.Kub
 		APIVersion: kubeletconfig.SchemeGroupVersion.String(),
 		Kind:       "KubeletConfiguration",
 	}
-	config.StaticPodPath = constants.ManifestsDirectory
+
+	if cfgSpec.EnableManifestsDirectory {
+		config.StaticPodPath = constants.ManifestsDirectory
+	}
```

Comment:
```suggestion
	} else {
	    config.StaticPodPath = ""
        }
```

this is required to make sure if manifest directory is disabled in the config, we update the config properly